src/main/java/org/openhab/tools/analysis/checkstyle/EshInfXmlCheck.java

```diff
@@ -0,0 +1,348 @@
+                        Map<String, File> configDescriptions = evaluateExpressionOnFile(xmlFile,
+                                CONFIG_DESCRIPTION_EXPRESSION);
+                        allConfigDescriptions.putAll(configDescriptions);
+                        break;
+                    }
+                    case BINDING_DIRECTORY: {
+                        schema = bindingSchema;
+                        // the allowed values are described in the binding XSD
+                        Map<String, File> configDescriptionRefs = evaluateExpressionOnFile(xmlFile,
+                                CONFIG_DESCRIPTION_REF_EXPRESSION);
+                        allConfigDescriptionRefs.putAll(configDescriptionRefs);
+                        Map<String, File> configDescriptions = evaluateExpressionOnFile(xmlFile,
+                                CONFIG_DESCRIPTION_EXPRESSION);
+                        allConfigDescriptions.putAll(configDescriptions);
+                        break;
+                    }
+                    case CONFIGURATION_DIRECTORY: {
+                        schema = configSchema;
+                        // the allowed values are described in the config description XSD
+                        Map<String, File> configDescriptions = evaluateExpressionOnFile(xmlFile,
+                                CONFIG_DESCRIPTION_EXPRESSION);
+                        allConfigDescriptions.putAll(configDescriptions);
+                        break;
+                    }
+                    default:
+                        // Other directories like l18n are allowed, but they are not object of this check, so they will
+                        // be skipped
+                        break;
+                }
+                if (schema != null) {
+                    validateXmlAgainstSchema(xmlFile, schema);
+                }
+            }
+        }
+    }
+
+    private void processBuildProperties(File file) throws CheckstyleException {
+        try {
+            buildPropertiesFile = parseBuildProperties(file);
+        } catch (CheckstyleException e) {
+            logger.error("Problem occurred while parsing the file {}", file.getPath(), e);
+        }
+    }
+
+    private Map<String, File> evaluateExpressionOnFile(File xmlFile, String xPathExpression)
+            throws CheckstyleException {
+        Map<String, File> collection = new HashMap<>();
+        NodeList nodes = getNodes(xmlFile, xPathExpression);
+
+        if (nodes != null) {
+            for (int i = 0; i < nodes.getLength(); i++) {
+                collection.put(nodes.item(i).getNodeValue(), xmlFile);
+            }
+        }
+        return collection;
+    }
+
+    private NodeList getNodes(File xmlFile, String expression) throws CheckstyleException {
+        Document document = parseDomDocumentFromFile(xmlFile);
+
+        XPathExpression xpathExpression = compileXPathExpression(expression);
+
+        NodeList nodes = null;
```

Comment:
Pity, the solutions I can find are worse :-(

http://stackoverflow.com/questions/30897704/initialize-nodelist-java#answer-30897953